framework/include/bcs/CoupledVarDirichletBC.h

```diff
@@ -0,0 +1,37 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "DirichletBCBase.h"
+
+/**
+ * Implements a Dirichlet BC where u=_coupled_var on the boundary.
+ */
+class CoupledVarDirichletBC : public DirichletBCBase
+{
+public:
+  static InputParameters validParams();
+
+  CoupledVarDirichletBC(const InputParameters & parameters);
+
+protected:
+  virtual Real computeQpValue() override;
+  virtual Real computeQpJacobian(unsigned int jvar);
+  virtual Real computeQpOffDiagJacobian(unsigned int jvar);
```

Comment:
these really should override. If they dont then they are likely not being called